### ISSUE TITLE
Fully annotated CoNLL-U examples in docs

### ISSUE THREAD

**Issue by martinpopel** [UniversalDependencies/docs#365]
I suggest almost* all examples in the documentation are fully annotated (all CoNLL-U columns) and allow the users to show the source CoNLL-U using the [tabs=yes](http://universaldependencies.org/visualization.html#editing) Brat feature.

I often encounter examples where I would like to see e.g. the features or other dependencies than those selected by someone as relevant for the example. Also in [the ellipsis documentation](http://universaldependencies.org/u/overview/enhanced-syntax.html#ellipsis), I would like to see how the empty nodes are stored in CoNLL-U. This can be partially guessed from [the CoNLL-U spec](http://universaldependencies.org/format.html#words-tokens-and-empty-nodes), which shows just the first three columns, so I don't know whether the empty nodes should have features etc. annotated and how to store enhanced dependencies to empty nodes.

*) I agree the examples should show/highlight only the relevant part. Brat shows lemmas, features and misc as tooltips so including them should not hurt the clarity. It would be nice if Brat could be configured to show upos tags only as tooltips (sometimes the upos tag is wider than the word and including tags makes the image bigger). The biggest issue is with the dependencies which are not considered relevant for the example. Maybe we could highlight the relevant dependencies in bold.

**Comment by dan-zeman** [comment]
While I agree it would be sometimes useful to have, I do not think it is doable with the manpower we currently have. Most examples are made up for the documentation, not taken from corpora, and the difference in effort needed to create a full CoNLL-U sentence (or just anything in CoNLL-U) vs. a simple "sdparse" excerpt, is significant. I only use the CoNLL-U format when I need to show something that cannot be shown using sdparse (or I do not know how).

**Comment by martinpopel** [comment]
One possible solution: use the [UDPipe web service](http://lindat.mff.cuni.cz/services/udpipe/run.php), select the English model, input plain-text sentence and copy-paste the output CoNLL-U (and check the svg image for errors in the automatic analysis). Of course, you need to post-edit it - the models are v1 so dobj->obj etc. is needed.

It would not be difficult to implement sdparse-to-conllu converter. I can help with this by implementing sdparse reader in [Udapi](http://udapi.github.io/) and possibly another block which adds the missing columns (features and misc) from UDPipe analysis.

**Comment by arademaker** [comment]
@Martinpopel what would be a sdparse-to-conllu converter ? In http://universaldependencies.org/visualization.html#editing   I saw that besides the sdparse we also have the conllx and conllu 'readers' right ?

**Comment by martinpopel** [comment]
> In http://universaldependencies.org/visualization.html#editing I saw that besides the sdparse we also have the conllx and conllu 'readers' right ?

There are two issues:
1. sdparse is probably a bit easier to enter manually than conllu (which is about the same complex as conllx, but conllx is not really suitable for UD). However, once you need the features, you need to use the `Token/TAG[Feat1=Val1|Feat2=Val2]` syntax (which is about the same complex as conllu, I think) and I am not aware of a way how to enter MISC, DEPS, empty nodes and multi-word tokens in the sdparse syntax. Also, if you enter sdparse, Brat currently cannot show the source as conllu, so Brat does not really work as a converter. Users want to know how CoNLL-U files should look like as this (unlike sdparse) is the primary format of UD and they need to follow when releasing/using their treebanks.
2. It will be quite some work to add the full annotation (especially features) for all the examples. This issue can be solved by using automatic annotation by UDPipe and just check if it is correct. This will also solve the first issue - we will use conllu input for Brat.

> what would be a sdparse-to-conllu converter ?

This would be a standalone tool implemented in [Perl](https://github.com/udapi/udapi-perl) or [Python](https://github.com/udapi/udapi-python).

**Comment by spyysalo** [comment]
I agree with @dan-zeman that the cost of switching to CoNLL-U for all examples is likely to be excessively high. I also don't think any automatic solution that has the potential to introduce errors is likely to be accepted, and I would oppose applying one to the parts of the documentation that I'm maintaining. As the `sdparse` format is arguably easier for people to read and write, removing it as an option would also increase the (already quite high) effort required to maintain examples.

I think a reasonable solution might be to provide CoNLL-U views of the examples, regardless of the format they have in the original source. If anyone is prepared to write a JS sdparse-to-conllu converter, I could incorporate such a view as an extra tab. However, note that there was some opposition to making the tabs active in all examples when I last suggested this, so it may be necessary to come up with a less visually intrusive way to access this functionality.

**Comment by martinpopel** [comment]
I see. I meant the automatic solution only as a pre-annotation, which *must* be checked by a human (this way I think it is less error-prone than annotating from scratch, where humans tend to forget which all features should be included etc). I didn't want to remover `sdparse` as an option. I am just not sure if it is really easier to read and write once all the info from CoNLL-U is included.

It would be great if Brat could provide CoNLL-U view even for examples originally written in sdparse. I like the tabs, but if a majority of UD core thinks it is visually intrusive, we could have there just three dots/lines (as the menu icon in Chrome/Firefox, etc.) as a pop-up menu. (An alternative would be to capture righ-clicks, but that's so unobtrusive that most users won't notice this feature:-).)

**Comment by spyysalo** [comment]
> I meant the automatic solution only as a pre-annotation, which *must* be checked by a human

This would be a good option to have, but a lot of work would be required to apply it to all examples:

    universaldependencies-docs smp$ find . -name '*.md' | xargs cat | egrep '^\~\~\~ .' | sort | uniq -c | sort -rn
    6469 ~~~ conllu
    2651 ~~~ sdparse

Making a systematic manual revision of thousands of examples in ~50 languages with documentation maintained by as many people would probably be months of work just for the person coordinating the effort.

I hate to be negative -- I think more complete examples would be great to have, and having the option of automatic pre-annotation for any UD language would be excellent (I'd use it!) -- but I don't think it's likely we'll be able to do this at the scale of the full documentation.

A CoNLL-U view of the information contained in the existing examples should be quite doable, though, and as I wrote previously, I'll be happy to work on the UI if someone can provide a JavaScript conversion of the SD format (sdparse) to CoNLL-U.

**Comment by dan-zeman** [comment]
```
  44508 ~~~ conllu
   5726 ~~~ sdparse
     13 ~~~ sdparse
      7 ~~~ sdparse
      2 ~~~ sdparse~~~
      2 ~~~ sdparse ~~~
      2 ~~~
      1 ~~~ sdparse
      1 ~~~ sdparse
      1 ~~~ conllx
      1 ~~~  conllu
      1 ~~~
```
